library/psa_crypto_hash.c

```diff
@@ -73,6 +73,25 @@ psa_status_t mbedtls_psa_hash_abort(
         case PSA_ALG_SHA_512:
             mbedtls_sha512_free(&operation->ctx.sha512);
             break;
```

Comment:
`mbedtls_md_type_from_psa_alg` in `md.c` is missing the SHA3 cases. This explains why OAEP and PSS test cases are failing. In both cases, this function is needed to select the algorithm for `mbedtls_rsa_set_padding`. For some legacy reason, `mbedtls_rsa_set_padding` succeeds despite having `hash_id=NONE` (maybe because you can, or could, set the mode and the hash separately?), but later `mbedtls_rsa_xxx_{oaep,pss}` sees that `ctx->hash_id` is invalid and bails out.

`mbedtls_md_psa_alg_from_type` is also missing that case. This would break things we don't support or don't test yet, such as using SHA3 in TLS, or possibly in X509 with USE_PSA_CRYPTO.

We have a test gap here. The `md_info` test in `test_suite_md` should test those metadata conversion functions.

Reply:
Thanks, I will sort this & improve the test coverage here